vad.py

```diff
@@ -53,14 +11,118 @@ def resize_to_min_size_(*tensors, dim = -1):
 			sliced = t.narrow(dim, 0, size)
 			t.set_(t.storage(), 0, sliced.size(), sliced.stride())
 
-class WebrtcSpeechActivityDetectionModel(nn.Module):
-	def __init__(self, aggressiveness):
+
+class PrimitiveVAD:
+	def __init__(self,
+	             device: str = 'cpu',
+	             kernel_size_smooth_silence: int = 4096,
+	             kernel_size_smooth_signal: int = 128,
+	             kernel_size_smooth_speaker: int = 4096,
+	             silence_absolute_threshold: float = 0.05,
+	             silence_relative_threshold: float = 0.2,
+	             eps: float = 1e-9,
+	             normalization_percentile: float = 0.9):
+		self.device = device
+		self.kernel_size_smooth_silence = kernel_size_smooth_silence
+		self.kernel_size_smooth_signal = kernel_size_smooth_signal
+		self.kernel_size_smooth_speaker = kernel_size_smooth_speaker
+		self.silence_absolute_threshold = silence_absolute_threshold
+		self.silence_relative_threshold = silence_relative_threshold
+		self.eps = eps
+		self.normalization_percentile = normalization_percentile
+		self.required_wrapper = torch.tensor
+		self.required_type = 'float32'
+
+	def detect(self, signal: shapes.BT, keep_intersections: bool = False) -> shapes.BT:
+		assert len(signal) <= 2
+		signal = signal.to(self.device)
+
+		padding = self.kernel_size_smooth_signal // 2
+		stride = 1
+		smoothed_for_diff = F.max_pool1d(signal.abs().unsqueeze(1), self.kernel_size_smooth_signal, stride=stride, padding=padding).squeeze(1)
+
+		padding = self.kernel_size_smooth_silence // 2
```

Comment:
if they apply to dilation, please move these lines down

Reply:
?